mobile/src/main/java/ch/epfl/sdp/mobile/ui/speech_recognition/SpeechRecognitionFilter.kt

```diff
@@ -0,0 +1,14 @@
+package ch.epfl.sdp.mobile.ui.speech_recognition
+
+/** Filtering interface for speech recognizer results */
+interface SpeechRecognitionFilter {
+  /**
+   * @param speeches list of recognized speeches
+   * @param confidencesScores assigned scores to each recognized speech
+   * @return filtered results
+   */
+  fun filterWords(
+      speeches: List<String>,
+      confidencesScores: List<Double>,
```

Comment:
You may want to use a dedicated data class to have a single list of `Result(val text: String, val score: Double)`.